app/server.go

```diff
@@ -484,7 +484,9 @@ func NewServer(options ...Option) (*Server, error) {
 		pluginsEnvironment.InitPluginHealthCheckJob(*s.Config().PluginSettings.Enable && *s.Config().PluginSettings.EnableHealthCheck)
 	}
 	s.AddConfigListener(func(_, c *model.Config) {
+		s.PluginsLock.RLock()
 		pluginsEnvironment := s.PluginsEnvironment
+		s.PluginsLock.RUnlock()
```

Comment:
Is this enough to prevent the race? This is a shallow copy of environment and the actual action we take here is to use one of the pointers in the struct.

Reply:
Yes, it does. The race is about accessing the `PluginsEnvironment` field from inside the Server struct.